### PR TITLE
feat: add value to the runtime crate

### LINKED ISSUE #13: feat: add value on runtime crate

### PR DESCRIPTION
**Pull request by algebraic-dev**

### CODE REVIEW

**Review by aripiprazole — APPROVED** [review]

File: crates/soft-runtime/src/value/freeable.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+use super::*;
+
+pub trait Freeable where Self : crate::value::tagged::Tagged {
+    const LAYOUT: Layout = Layout::new::<Self>();
+
+    fn free(ptr: Ptr<Self>) {
+        unsafe {
+            dealloc(ptr.0 as *mut u8, Self::LAYOUT)
```
Comment (marcodsl) [inline crates/soft-runtime/src/value/freeable.rs]: 🌶️

File: crates/soft-runtime/src/value/pointer.rs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        }
+    }
+}
+
+impl Str {
+    pub fn new(str: String) -> Str {
+        Str(Box::leak(str.into_boxed_str()))
+    }
+
+    pub fn free(&self) {
```
Comment (algebraic-dev) [inline crates/soft-runtime/src/value/pointer.rs]: Move this function

**Review by noghartt — APPROVED** [review]

File: crates/soft-runtime/src/value/pointer.rs [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    fn test_vector() {
+        let va = Int::new(42).into();
+        let vb = Str::new("eu amo a gabii".to_string()).into();
+        let cons1: Value = Cons::new(va, vb).into();
+        let cons: Value = Cons::new(cons1, cons1).into();
+
+        let vector: Value = Vector::new(vec![cons, cons]).into();
+
+        let r = vector.classify();
+        assert_eq!(r.to_string(), "[((42 . \"eu amo a gabii\") 42 . \"eu amo a gabii\") ((42 . \"eu amo a gabii\") 42 . \"eu amo a gabii\")]");
```
Comment (wrongbyte) [inline crates/soft-runtime/src/value/pointer.rs:243]: fofo

**Review by wrongbyte — APPROVED** [review]

File: crates/soft-runtime/src/value/pointer.rs (hunk 3 above)
Comment (noghartt) [reply crates/soft-runtime/src/value/pointer.rs:243]: MUITO FOFO

**Pull request merged**